system_baseline/validators.py

```diff
@@ -45,6 +46,17 @@ def check_for_empty_name_values(facts):
         elif "value" in fact and not fact["value"]:
             raise FactValidationError("value for %s cannot be empty" % fact["name"])
 
+def check_for_invalid_whitespace_name_values(facts):
+    """
+    check if any name or values have invalid whitespace at beginning and end; raises an exception.
+    """
+    for fact in facts:
+        if "values" in fact:
+            check_for_invalid_whitespace_name_values(fact["values"])
+        if "name" in fact and re.search(r'^\s|\s$', fact["name"]):
+            raise FactValidationError("fact name cannot have leading or trailing whitespace")
+        elif "value" in fact and re.search(r'^\s|\s$', fact["value"]):
+            raise FactValidationError("value for %s cannot have leading or trailing whitespace" % fact["name"])
```

Comment:
There are a ton of ways to solve this problem, and this works!

However, it's preferable to not use a regex if possible to make the code more maintainable. Here is an example of how to do it without regexes:

```python
import string

starts_with_space = " micjohns"
ends_with_space = "micjohns "
ok_string = "micjohns"
empty_string = ""
lol_string = None

def check_spaces(value):
    if value and value[0] in string.whitespace:
        print("string '%s' starts with a space" % value)
    elif value and value[-1] in string.whitespace:
        print("string '%s' ends with a space" % value)

check_spaces(starts_with_space)
check_spaces(ends_with_space)
check_spaces(ok_string)
check_spaces(empty_string)
check_spaces(lol_string)
```

Reply:
you could also do something like

```python
if value != value.lstrip():
    print("value starts with whitespace")
elif value != value.rstrip():
    print("value ends with whitespace")
```
lstrip and rstrip remove leading/trailing white space